spirv_msl.cpp

```diff
@@ -372,7 +405,9 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage)
 			// Copy the variable location from the original variable to the member
 			auto &dec = meta[p_var->self].decoration;
 			uint32_t locn = dec.location;
-			set_member_decoration(ib_type_id, ib_mbr_idx, DecorationLocation, locn);
+			if (get_decoration_mask(p_var->self) == DecorationLocation) {
```

Comment:
This test should be `& (1ull << DecorationLocation)` instead of `== DecorationLocation`. And I'm curious...under what conditions are you not getting the location from SPIR-V?